test/OrderByDescending.spec.js

```diff
@@ -0,0 +1,29 @@
+const expect = require('chai').expect;
```

Comment:
Comments from `test/OrderBy.spec.js` apply to this file as well.

Reply:
x => x.v1

Reply:
With the logic. I'm not too sure how the implement the sort. I tried using the keySelector function to help with the compareFunction (but the list is unaffected) i.e.,


```
// Making a call like [{ v1: 2, v2: 'b' }, { v1: 3, v2: 'a' }].OrderBy(x => x.v2);

// Below is the code

function OrderBy(keySelector, comparer = _defaultComparer) {
  if (keySelector == null)
    throw 'Argument Null Exception - keySelector is null.';
  if (this === undefined) throw 'Undefined value exception!';
  if (this === null && typeof this === 'object') throw 'Null value exception!';
  return this.reduce(e => {
    {
      return JSON.parse(JSON.stringify(this)).sort(function(a, b) {
        return comparer * Math.abs(a[keySelector(e)] - b[keySelector(e)]);
      });
    }
  });
}
```